### ISSUE TITLE
Navigating the document should change URL

### ISSUE THREAD

**Issue by viktor-yakubiv** [oacore/reader#15]
A click on a link either on sidebar navigation or PDF document navigation should change the URL of the page by adding a hash.

## Why?

Click on any link expects a navigation transition, link to this page expects a transition to corresponding section. Any transition must be represented to the URL.

Also, it is good to have this feature for the case when a user want's to share a particular section of the document. He/She should be able to:
- copy the URL from the address bar if he already navigated to the section
- click on _Copy the link_ in the context menu of the link in the sidebar or PDF document navigation
